[Visualize] Centralize dependencies in kibana_services.ts

maryia-lapata opened the pull request:
## Summary

This PR moves all imports of other plugins into a central file to prepare for shimming the plugin.
Installing the shim and getting rid of the local angular router will be done in a separate PR.

[Saved visualizations folder](https://github.com/elastic/kibana/tree/master/src/legacy/core_plugins/kibana/public/visualize/saved_visualizations) is out of scope because there is a plan to migrate it.

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- ~~[ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~
- ~~[ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
- ~~[ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
- ~~[ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios~~
- ~~[ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

- ~~[ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~
- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/6939/)
* Commit: 192b88ceb031a6d8cc548db6fa153dcc6612eb40

elasticmachine commented:
Pinging @elastic/kibana-app (Team:KibanaApp)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/6958/)
* Commit: abfdc2aa48fc31067d5345ab030101115bc58955

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/6975/)
* Commit: 57a1115a7d416db1109bfa8862a51538139ab617

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/7005/)
* Commit: 72fd4acfd20349febc5c154a12ae045085d4f467

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/7301/)
* Commit: 65ef3d063aab523b9795430596471217f739b58e

TinaHeiligers commented:
@maryia-lapata @flash1293 , I was going through this PR as part of getting familiar with np migrations efforts and noticed a `chrome` import from `ui/chrome` remaining in a test [here](https://github.com/elastic/kibana/pull/49293/files?file-filters%5B%5D=.tsx#diff-1b82063e3b22806bd4bf5a4bb7aff17eR23). `chrome` doesn't seem to be used in the file. Does `'../../kibana_services'` [mock](https://github.com/elastic/kibana/pull/49293/files?file-filters%5B%5D=.tsx#diff-1b82063e3b22806bd4bf5a4bb7aff17eR25) need access to `chrome`?

maryia-lapata commented:
@flash1293 I moved `data` import to `kibana_services.ts` as well as imports from `plugins/embeddable/public`.

@TinaHeiligers nice catch! I removed a `chrome` import from `ui/chrome`.

Thank you.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/7497/)
* Commit: d66b1adc46bb15a925f3c4b16d21a01044b67b1f

maryia-lapata commented on x-pack/legacy/plugins/canvas/canvas_plugin_src/expression_types/embeddable_types.ts:
```diff
@@ -6,7 +6,7 @@
 // @ts-ignore
 import { MAP_SAVED_OBJECT_TYPE } from '../../../maps/common/constants';
-import { VISUALIZE_EMBEDDABLE_TYPE } from '../../../../../../src/legacy/core_plugins/kibana/public/visualize/embeddable';
+import { VISUALIZE_EMBEDDABLE_TYPE } from '../../../../../../src/legacy/core_plugins/kibana/public/visualize/embeddable/constants';
```
@elastic/kibana-canvas I had to change this import, to ensure, that just this const is imported, and not all of centralized dependencies from Visualize plugin.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/7508/)
* Commit: 39e64e3c0a90b06d72b0cbd51cc606cc080fa36d

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/7584/)
* Commit: 8dd28f785f4b0f7cfc686671466b789258f36b84

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/7803/)
* Commit: a6149b54261ba2b232bfdc13358b956836f15846
